src/frontend/components/UI/AccountDropdown/index.tsx

```diff
@@ -1,40 +1,42 @@
 import { Menu } from '@mantine/core'
-import React, { useContext } from 'react'
+import React from 'react'
+import walletStore from 'frontend/state/WalletState'
+import { observer } from 'mobx-react-lite'
+
 import Wallet from 'frontend/components/UI/Wallet'
 import styles from './index.module.scss'
 import { NavLink } from 'react-router-dom'
 import onboardingStore from 'frontend/store/OnboardingStore'
 import { useTranslation } from 'react-i18next'
-import ContextProvider from 'frontend/state/ContextProvider'
 
-function NavigationMenuItem({ label, to }: { label: string; to: string }) {
-  const { showMetaMaskBrowserSidebarLinks } = useContext(ContextProvider)
+function NavigationMenuItem({
+  label,
+  to,
+  showWalletLinks
+}: {
+  label: string
+  to: string
+  showWalletLinks: boolean
+}) {
   return (
     <Menu.Item
       className={styles.menuItem}
-      id={
-        showMetaMaskBrowserSidebarLinks
-          ? 'topMenuItemWalletDropdown'
-          : undefined
-      }
+      id={showWalletLinks ? 'topMenuItemWalletDropdown' : undefined}
     >
       <NavLink
         to={to}
-        id={
-          showMetaMaskBrowserSidebarLinks
-            ? 'topElementWalletDropdown'
-            : undefined
-        }
+        id={showWalletLinks ? 'topElementWalletDropdown' : undefined}
       >
         <div className={`body ${styles.itemContents}`}>{label}</div>
       </NavLink>
     </Menu.Item>
   )
 }
 
-export default function AccountDropdown() {
+const WalletDropdown: React.FC = observer(() => {
   const { t } = useTranslation()
-  const { showMetaMaskBrowserSidebarLinks } = useContext(ContextProvider)
+  const showWalletLinks = walletStore.isConnected
```

Comment:
this will show the wallet links when MetaMask mobile/wallet connect wallets are connected. metamask snaps and metamask home should only be shown if MetaMask extension is connected

Reply:
so how should I fix that ?

Reply:
this comment is about the same issue https://github.com/HyperPlay-Gaming/hyperplay-desktop-client/pull/634#discussion_r1394800406

Reply:
Ok so now I changed it to only show MM portfolio and Snaps if it is MM wallet.
If is another wallet it will only show the Change Wallet button.